### DISCUSSION TITLE
Validate question/question sets while submitting for review/publish

### DISCUSSION THREAD

**Discussion by pallakartheekreddy** [Sunbird-inQuiry/Community#66] [Improvements]
The create API in inQuiry allows creation of a question object with minimal data (e.g.name, code, mimeType, primaryCategory) without the actual question body/option etc.,
In the Sunbird Bootcamp, we observed that when such a question is allowed to be sent for review/publish, it fails the validation in the publish pipeline, Since publish pipeline has validation checks for requires certain metadata.

Hence, we plan to resolve this issues as part of the 5.6 release, by having validation for the question metadata before sending it for review/publish. Please find JIRA ticket and more details in link.
https://project-sunbird.atlassian.net/browse/IQ-300

@krgauraw @PoojaJManjooran @surendrasinghs @sajeshkayyath @panduankani @snehar-nd @israr-trigyn @Xitija @vishalisakar 
@mansurskTarento @BharathTarento @adithyadinesh0412 @Vinod-V3 @shreyasb22 @shashankpace @shwetha-rao13

**Comment by krgauraw** [comment]
@pallakartheekreddy As part of IQ-300, we have added validation check in question/questionset review and publish api. create api's will continue to create object with minimal data.

cc: @PoojaJManjooran @surendrasinghs @sajeshkayyath @panduankani @snehar-nd @israr-trigyn @Xitija @vishalisakar
@mansurskTarento @BharathTarento @adithyadinesh0412 @Vinod-V3 @shreyasb22 @shashankpace @shwetha-rao13
